dashboard/app/models/user.rb

```diff
@@ -952,6 +1007,14 @@ def recent_activities(limit = 10)
     self.activities.order('id desc').limit(limit)
   end
 
+  def can_pair?
+    !sections_as_student.empty?
+  end
+
+  def can_pair_with?(other_user)
+    sections_as_student.any? {|section| other_user.sections_as_student.include? section}
```

Comment:
Should there be a check in `can_pair_with` to make sure that the current user and `other_user` are distinct?